### PR TITLE
Changes required for macOS 10.15

### PR DESCRIPTION
**Pull request by sivel**
##### SUMMARY
Changes required for macOS 10.15

##### ISSUE TYPE
- Bugfix Pull Request

##### ADDITIONAL INFORMATION

```paste below

```
Ref of upstream timezone change: https://github.com/ansible-collections/community.general/pull/945

### CODE REVIEW

**Comment by samdoran** [comment]
I took care of the skips and adding the symlink for `macos.sh` in https://github.com/ansible/ansible/pull/71849

File: test/integration/targets/prepare_http_tests/tasks/default.yml
Comment (mattclay) [inline test/integration/targets/prepare_http_tests/tasks/default.yml]: This looks like it should use the running Python version instead of being set to `3.8`.

Comment (sivel) [reply test/integration/targets/prepare_http_tests/tasks/default.yml]: Updated to be fully dynamic

**Review by mattclay — APPROVED** [review]
Looks good. We'll want to wait to merge this until @samdoran resolves the issues with reliable init on the new 10.15 images.

**Comment by samdoran** [comment]
I have gotten the kinks worked out of the `init` in the macOS 10.15 image and deployed it to the hosts. I kicked off the jobs again. 🤞

I'll review this tomorrow.

File: test/integration/targets/pip/tasks/main.yml
Comment (samdoran) [inline test/integration/targets/pip/tasks/main.yml:29]: Should this be a specific version of `wheel` to avoid spontaneous test breakage?

File: test/integration/targets/setup_paramiko/install-MacOSX-10-python-3.yml
Comment (samdoran) [inline test/integration/targets/setup_paramiko/install-MacOSX-10-python-3.yml:3]: Same thing here. Do we want to pin this to a specific version?

File: test/support/integration/plugins/modules/timezone.py
Comment (samdoran) [inline test/support/integration/plugins/modules/timezone.py:129]: Since this is a module change, should this be in a separate PR so it can be backported?

**Review by samdoran — APPROVED** [review]
Just a few minor questions. Thanks for doing this.

File: test/integration/targets/setup_paramiko/install-MacOSX-10-python-3.yml
Comment (sivel) [reply test/integration/targets/setup_paramiko/install-MacOSX-10-python-3.yml:3]: It's not pinned in any of the other files for this target either. I suppose we could, but I'm going to avoid that in this PR.

File: test/support/integration/plugins/modules/timezone.py
Comment (sivel) [reply test/support/integration/plugins/modules/timezone.py:129]: I'd rather just leave it here, and wait until the upstream PR is merged before backporting. We should probably not cherry-pick this anyway until the upstream PR is merged: https://github.com/ansible-collections/community.general/pull/945

2.9 tests will likely have to be updated further than devel if we backport the macos tests.

Comment (samdoran) [reply test/support/integration/plugins/modules/timezone.py:129]: I think the intention was to only backport this to `stable-2.10`. We'll continue testing `stable-2.8` and `stable-2.9` with OS X 10.11.

Comment (sivel) [reply test/support/integration/plugins/modules/timezone.py:129]: I'd just leave this as is then.  This is just testing support.

**Pull request merged**
